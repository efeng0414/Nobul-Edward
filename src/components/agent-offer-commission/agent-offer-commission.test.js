import React from "react";
import renderer from "react-test-renderer";
import AgentOfferCommission from "./index";
import calculateCommision from "../../../core/utilities/calculate-commission";
import { IntlProvider, intlShape } from "react-intl";

describe("AgentOfferCommission", () => {
  const componentProps = {
    value: 1.2,
    priceLow: 123,
    priceHigh: 456,
    intl: global.getIntl(),
    commissionType: "buy",
    onChange: () => {}
  };

  const componentJSX = <AgentOfferCommission {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.intl}>{componentJSX}</IntlProvider>
    );

    const componentSnapshot = renderer.create(snapshotJSX).toJSON();
    it("should match stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });
  describe("Container mount", () => {
    const contextObject = {
      context: { intl: componentProps.intl },
      childContextTypes: { intl: intlShape.isRequired }
    };

    const functionKeysToReset = ["handleCommissionChange", "calculateRange"];
    const componentPrototypeResets = functionKeysToReset.reduce(
      global.createComponentPrototypeResets({
        component: AgentOfferCommission
      }),
      {}
    );

    const componentMount = global.Enzyme.shallow(componentJSX, contextObject);
    beforeEach(() => {
      functionKeysToReset.map(
        global.resetPropertyComponentProperty({
          component: AgentOfferCommission,
          componentMount: componentMount,
          componentPrototypeResets
        })
      );
    });
    const mockCommission = 4.9;

    it("should call calculateRange function on button click", () => {
      const calculateRangeSpy = jest.fn();
      componentMount.instance().calculateRange = calculateRangeSpy;
      componentMount.setProps({ a: 1, b: 2, c: 3 });
      componentMount.find("Button").simulate("click");
      expect(calculateRangeSpy).toHaveBeenCalled();
    });

    it("it should check if calculateCommission function is working correctly", () => {
      const mockPrice = 2.3;
      const mockCommission = 3.2;
      expect(calculateCommision(mockPrice, mockCommission)).toBe("0.07");
    });

    it("should set state when calculateRange function is called", () => {
      componentMount.setState({ commission: mockCommission });
      componentMount.instance().calculateRange();
      const expectedCalculatedLow = calculateCommision(
        componentProps.priceLow,
        mockCommission
      );
      const expectedCalculatedHigh = calculateCommision(
        componentProps.priceHigh,
        mockCommission
      );

      expect(componentMount.state().calculatedLow).toBe(expectedCalculatedLow);
      expect(componentMount.state().calculatedHigh).toBe(
        expectedCalculatedHigh
      );
    });

    it("should call handleCommissionChange function on Input Change", () => {
      const handleCommissionSpy = jest.fn();
      componentMount.instance().handleCommissionChange = handleCommissionSpy;
      componentMount.setProps({ b: 1 });
      componentMount.find("InputNumber").simulate("change");
      expect(handleCommissionSpy).toHaveBeenCalled();
    });

    it("should setState of commission when handleCommissionChange is called", () => {
      const fakeMockCommissionState = 1;
      componentMount.setState({ commission: fakeMockCommissionState });
      componentMount.instance().handleCommissionChange(mockCommission);
      expect(componentMount.state("commission")).toBe(mockCommission);
    });

    it("should call this.props.onChange function when handleCommissionChange is called", () => {
      const mockOnChange = jest.fn();
      componentMount.setProps({ onChange: mockOnChange });
      componentMount.instance().handleCommissionChange(mockCommission);
      expect(mockOnChange).toHaveBeenCalled();
    });
  });
});
