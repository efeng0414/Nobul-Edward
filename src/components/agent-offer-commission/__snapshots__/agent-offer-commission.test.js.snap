// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentOfferCommission Component Snapshot should match stored snapshot 1`] = `
<div
  className="offer-commission"
>
  <div
    className="offer-commission__values"
  >
    <div
      className="ant-input-number"
      style={Object {}}
    >
      <div
        className="ant-input-number-handler-wrap"
      >
        <span
          aria-disabled={false}
          aria-label="Increase Value"
          className="ant-input-number-handler ant-input-number-handler-up "
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          unselectable="unselectable"
        >
          <span
            className="ant-input-number-handler-up-inner"
            onClick={[Function]}
            unselectable="unselectable"
          />
        </span>
        <span
          aria-disabled={false}
          aria-label="Decrease Value"
          className="ant-input-number-handler ant-input-number-handler-down "
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          unselectable="unselectable"
        >
          <span
            className="ant-input-number-handler-down-inner"
            onClick={[Function]}
            unselectable="unselectable"
          />
        </span>
      </div>
      <div
        aria-valuemin={-9007199254740991}
        aria-valuenow={1.2}
        className="ant-input-number-input-wrap"
        role="spinbutton"
      >
        <input
          autoComplete="off"
          className="ant-input-number-input"
          min={-9007199254740991}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseUp={[Function]}
          required={false}
          step={0.25}
          value="1.20"
        />
      </div>
    </div>
    <span
      className="offer-commission__divider"
    >
      %
    </span>
    <div
      className="offer-commission__range"
    >
      1.48
    </div>
    <span
      className="offer-commission__divider"
    >
      -
    </span>
    <div
      className="offer-commission__range"
    >
      5.47
    </div>
    <span
      className="offer-commission__divider"
    >
      <span
        className="currency"
      >
        <span
          className="currency__identifier"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class='currency__code'>CA</span>$",
            }
          }
        />
      </span>
    </span>
  </div>
  <button
    className="ant-btn offer-commission__button ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <span>
      Calculate
    </span>
  </button>
</div>
`;
